components/fonts/shaper.rs

```diff
@@ -414,12 +414,28 @@ impl Shaper {
                 })
             }
 
+            if servo_config::pref!(layout_variable_fonts_enabled) {
+                for (tag, value) in &options.variation_settings {
+                    let variations = &[hb_variation_t {
+                        tag: *tag,
+                        value: value.0,
+                    }];
+
+                    hb_font_set_variations(
+                        self.hb_font,
+                        variations.as_ptr(),
+                        variations.len() as u32,
+                    );
+                }
+            }
+
```

Comment:
Assuming this is enough to prevent the incorrect results, instead of a setting I think it makes sense to just conditionally compile this for Freetype platforms. I'm not sure a runtime setting is the way to go here since you should never turn it on and off...it depends totally on your platform.

Reply:
With a `cfg`-condition you would need to repeatedly update all the conditions as support for more platforms is added. And missing one could cause confusing results, thats why I used a runtime preference.

Reply:
Just to clarify, the benefit is that platforms get variation support as soon as it is implemented. This isn't typically something that web engines allow emedders to configure.

Reply:
I'll investigate if i can hook up something like `supports_variations` to stylo.

Reply:
Maybe it's a bit too much work compared to just adding support for variations on Windows and macOS. I'll see if I can do that here so that we don't have to spend so much time thinking about configurability.

Reply:
I think it's fine to keep the preference now and try to remove it in the next couple weeks.